### PR TITLE
uhcp: check maximum for prefix length

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Fixes https://github.com/RIOT-OS/RIOT/issues/15574.

### Testing procedure
See https://github.com/RIOT-OS/RIOT/issues/15574 for steps to reproduce and confirm that with this fix the issue is fixed.

### Issues/PRs references
Fixes https://github.com/RIOT-OS/RIOT/issues/15574.

### CODE REVIEW

File: sys/net/application_layer/uhcp/uhcp.c [hunk 1]
```diff
@@ -98,8 +98,8 @@ void uhcp_handle_push(uhcp_push_t *req, uint8_t *src, uint16_t port, uhcp_iface_
     char prefix_str[INET6_ADDRSTRLEN];
     inet_ntop(AF_INET6, src, addr_str, INET6_ADDRSTRLEN);
     uint8_t prefix[IN6ADDRSZ] = { 0 };
-    if (req->prefix_len == 0) {
-        LOG_ERROR("uhcp_handle_push(): prefix length 0\n");
+    if ((req->prefix_len == 0) || (req->prefix_len > (IN6ADDRSZ << 3))) {
```
Comment (miri64) [inline sys/net/application_layer/uhcp/uhcp.c:101]: I chose this notation to be in line with the notation below: `prefix` is of size `IN6ADDRSZ`, there are `prefix_bytes` bytes copied to `prefix` which is generated by bit-shifting `req->prefix_len` (+ 7 to round up) three bits to the right. So `req->prefix_len` must not be larger than `INADDRSZ` bit-shifted three bits to the left. Or to state it less complicated, `req->prefix_len` must be greater or equal to `INADDRSZ * 8` (bits), or 128...

**Comment by miri64** [comment]
@nmeum can you check if this fixes potential buffer overflows now?

**Review by kaspar030 — APPROVED** [review]
ACK ~~(once the comment promised offline is in :) )~~

**Comment by miri64** [comment]
Squashed as requested offline.

**Comment by nmeum** [comment]
> @nmeum can you check if this fixes potential buffer overflows now?

This is always a bit tricky for me to test. I am currently busy with other things, not sure if I manage to get around to it this year. Hope to revisit this eventually though.

**Comment by miri64** [comment]
Thanks for the review @kaspar030

**Comment by miri64** [comment]
> This is always a bit tricky for me to test. I am currently busy with other things, not sure if I manage to get around to it this year. Hope to revisit this eventually though.

Hope you won't find anything this time :-)

**Pull request merged**